radix-engine/src/blueprints/pool/one_resource_pool/blueprint.rs

```diff
@@ -569,18 +577,20 @@ impl OneResourcePoolBlueprint {
         pool_units_total_supply: Decimal,
         pool_resource_reserves: Decimal,
         pool_resource_divisibility: u8,
-    ) -> Decimal {
+    ) -> Result<Decimal, RuntimeError> {
         let amount_owed = pool_units_to_redeem
             .safe_mul(pool_resource_reserves)
-            .unwrap()
+            .ok_or_else(|| OneResourcePoolError::DecimalOverflowError)?
             .safe_div(pool_units_total_supply)
-            .unwrap();
+            .ok_or_else(|| OneResourcePoolError::DecimalOverflowError)?;
```

Comment:
Two minor comments about this: It might be better to use `.and_then` here + `.ok_or`. This will transform this code from:

```rust
let amount_owed = pool_units_to_redeem
    .safe_mul(pool_resource_reserves)
    .ok_or_else(|| OneResourcePoolError::DecimalOverflowError)?
    .safe_div(pool_units_total_supply)
    .ok_or_else(|| OneResourcePoolError::DecimalOverflowError)?;
```

To be:

```rust
let amount_owed = pool_units_to_redeem
    .safe_mul(pool_resource_reserves)
    .and_then(|value| value.safe_div(pool_units_total_supply))
    .ok_or(OneResourcePoolError::DecimalOverflowError)?
```

Reply:
good point!